src/combineReducers.lua

```diff
@@ -1,22 +1,32 @@
 --[[
 	Create a composite reducer from a map of keys and sub-reducers.
 ]]
-local function combineReducers(map)
-	return function(state, action)
-		-- If state is nil, substitute it with a blank table.
-		if state == nil then
-			state = {}
-		end
-
-		local newState = {}
-
-		for key, reducer in pairs(map) do
-			-- Each reducer gets its own state, not the entire state table
-			newState[key] = reducer(state[key], action)
-		end
-
-		return newState
-	end
+
+local _actions = require(script.Parent.types.actions)
+local _reducers = require(script.Parent.types.reducers)
+
+type AnyAction = _actions.AnyAction
+
+export type Reducer<State = any, Action = AnyAction> = _reducers.Reducer<State, Action>
+
+local function combineReducers<State>(map): Reducer<State>
+	return (
+		function(state, action)
```

Comment:
maybe something like this and eliminate the hard cast?
```suggestion
		function<S>(state: S?, action: AnyAction): S
```

Reply:
I tried a few things, but I settled on the any cast. I think this needs better generic type constraint support on the Luau side, otherwise you end up with some "unable to convert <insert some table type here> to State" errors, even when using the pseudo-constraint trick you showed me.